apis/rbac/v1alpha1/types.go

```diff
@@ -0,0 +1,302 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// ApplicationParameters parameters for an application.
+type ApplicationParameters struct {
+	// AvailableToOtherTenants - Whether the application is available to other tenants.
+	// +optional
+	// +immutable
+	AvailableToOtherTenants *bool `json:"availableToOtherTenants,omitempty"`
+
+	// DisplayName - The display name of the application.
+	// +optional
+	// +immutable
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// Homepage - The home page of the application.
+	// +optional
+	// +immutable
+	Homepage *string `json:"homepage,omitempty"`
+
+	// IdentifierUris - A collection of URIs for the application.
+	// +optional
+	// +immutable
+	IdentifierUris *[]string `json:"identifierUris,omitempty"`
+}
+
+// An ApplicationSpec defines the desired state of an Application.
+type ApplicationSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+
+	// ApplicationParameters parameters for an application.
+	ApplicationParameters `json:",inline"`
```

Comment:
Per Crossplane [API conventions](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md), this field should be under `spec.forProvider` and not inlined.  For an example, please see:
https://github.com/crossplane/provider-azure/blob/1c0702110ed45feabe2af15a0478e753d1c34d4e/apis/database/v1beta1/configuration_types.go#L103